src/main/java/com/target/studentinfo/service/StudentService.java

```diff
@@ -0,0 +1,71 @@
+package com.target.studentinfo.service;
+
+import com.target.studentinfo.dto.mapper.StudentMapper;
+import com.target.studentinfo.dto.request.StudentRequest;
+import com.target.studentinfo.dto.response.StudentResponse;
+import com.target.studentinfo.model.Student;
+import com.target.studentinfo.respository.StudentRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+
+public class StudentService {
+    private final StudentRepository studentRepository;
+
+    public StudentService(StudentRepository studentRepository) {
+        this.studentRepository = studentRepository;
+    }
+
+    public List<Student> getAllStudentInfo(){
+
+        return studentRepository.findAll();
+    }
+
+    public Optional<Student> getStudentById(UUID id){
+        return studentRepository.findById(id);
+    }
+
+    public Student addNewStudent(Student student){
+        return studentRepository.save(student);
+    }
+
+    public Student updateStudent(Student student) {
+        return studentRepository.save(student);
+    };
+
+    public void deleteStudent(UUID id) {
+        studentRepository.deleteById(id);
+    };
+
+    public StudentResponse addNewStudent(StudentRequest studentRequest) {
+
+        Student student = StudentMapper.MAPPER.fromReqToModel(studentRequest);
+        studentRepository.save(student);
+        return StudentMapper.MAPPER.fromModelToResponse(student);
+    };
```

Comment:
These mapper transformations should happen in the controller layer. The service layer should only deal with `Student` domain object. Transforming `from studentRequest` and `to studentResponse` should lie in controller.